### PR TITLE
Fix how we create and delete certificates in the log

### PR DESCRIPTION
**Pull request by ashamis**
This enables us to properly run more than 256 pre_prepare batches.

### CODE REVIEW

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

File: samples/apps/smallbank/clients/small_bank_client.cpp [hunk 1]
```diff
@@ -232,8 +232,8 @@ class SmallBankClient : public Base
           "No result in verification response: " + response.dump());
       }
 
-      auto expected_balance = balance_it->get<size_t>();
```
Comment (achamayou) [inline samples/apps/smallbank/clients/small_bank_client.cpp]: Why?

File: src/consensus/pbft/libbyz/Log.h [hunk 2]
```diff
@@ -117,28 +92,22 @@ template <class T>
...
-  int i = head;
-  int max = new_head;
-  if (new_head - head >= max_size)
-  {
-    i = 0;
-    max = max_size;
-  }
-
-  for (; i < max; i++)
+  for (auto it = std::begin(elems); it != std::end(elems);)
```
Comment (achamayou) [inline src/consensus/pbft/libbyz/Log.h]: elems.being() and elems.end()

File: src/consensus/pbft/libbyz/parameters.h [hunk 3]
```diff
@@ -12,7 +12,7 @@ const int Max_num_replicas = 32;
 // Interval in sequence space between "checkpoint" states, i.e.,
 // states that are checkpointed and for which Checkpoint messages are
 // sent.
-const int checkpoint_interval = 256;
```
Comment (achamayou) [inline src/consensus/pbft/libbyz/parameters.h]: How does this relate to sig_max_tx/ms? Is the batch size for the pre-prepares?

Comment (ashamis) [reply src/consensus/pbft/libbyz/parameters.h]: This is how often we flush our internal data structures. This all changes will Olga's work of when we recover the ledger.

File: src/consensus/pbft/libbyz/Client_proxy.h [hunk 4]
```diff
@@ -395,6 +398,9 @@ void ClientProxy<T, C>::retransmit()
       }
     }
 
+    LOG_INFO << "Client_proxy retransmitting request, rid:"
```
Comment (olgavrou) [inline src/consensus/pbft/libbyz/Client_proxy.h]: DEBUG or TRACE?

**Comment by cimetrics** [comment]
images

**Pull request merged**